### PR TITLE
[release-v1.9] Bump x/net to 0.17

### PR DESCRIPTION
**Pull request by pierDipi**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -66,15 +66,15 @@ require (
 	go.uber.org/atomic v1.9.0 // indirect
 	go.uber.org/automaxprocs v1.4.0 // indirect
 	go.uber.org/multierr v1.8.0 // indirect
-	golang.org/x/mod v0.6.0-dev.0.20220419223038-86c51ed26bb4 // indirect
-	golang.org/x/net v0.7.0 // indirect
+	golang.org/x/mod v0.8.0 // indirect
+	golang.org/x/net v0.10.0 // indirect
```
Comment (matzew) [inline go.mod:70]: 0.10.0 ? :-)

Comment (pierDipi) [reply go.mod:70]: hmm, not sure what happened

Comment (pierDipi) [reply go.mod]: https://github.com/openshift-knative/eventing-istio/pull/76/commits/40f89fcc6671a2d3df5fb2448b98aee3c4a7b324

Comment (pierDipi) [reply go.mod]: The actual vendor code is the same apparently

**Pull request merged**